midend/interpreter.cpp

```diff
@@ -727,6 +727,18 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         auto li = l->to<SymbolicInteger>();
         clone->expr = li->constant;
         DoConstantFolding cf(refMap, typeMap);
+        if (auto cast = expression->to<IR::Cast>()) {
+            if (cast->destType->is<IR::Type_Bits>()) {
+                const IR::Constant* constant;
+                if (clone->expr) {
```

Comment:
I don't think this does what you expect it does.
It compares the expr pointer to nullptr. It does not compare the value of the expression to true or false.